test/widgets/subscription_list_test.dart

```diff
@@ -0,0 +1,146 @@
+    final initialSnapshot = eg.initialSnapshot(
+      subscriptions: subscriptions,
+      streams: streams,
+      unreadMsgs: unreadMsgs,
+    );
+    await testBinding.globalStore.add(eg.selfAccount, initialSnapshot);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: GlobalStoreWidget(
+          child: PerAccountStoreWidget(
+            accountId: eg.selfAccount.id,
+            child: const SubscriptionListPage()))));
+
+    // global store, per-account store
+    await tester.pumpAndSettle();
+  }
+
+  bool isPinnedHeaderInTree() {
+    return find.text('Pinned').evaluate().isNotEmpty;
+  }
+
+  bool isUnpinnedHeaderInTree() {
+    return find.text('Unpinned').evaluate().isNotEmpty;
+  }
+
+  int getItemCount() {
+    return find.byType(SubscriptionItem).evaluate().length;
+  }
+
+  testWidgets('smoke', (tester) async {
+    await setupStreamListPage(tester, subscriptions: []);
+    check(getItemCount()).equals(0);
+    check(isPinnedHeaderInTree()).isFalse();
+    check(isUnpinnedHeaderInTree()).isFalse();
+  });
+
+  testWidgets('basic subscriptions', (tester) async {
+    await setupStreamListPage(tester, subscriptions: [
+      eg.subscription(eg.stream(streamId: 1), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 2), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 3), pinToTop: false),
+    ]);
+    check(getItemCount()).equals(3);
+    check(isPinnedHeaderInTree()).isTrue();
+    check(isUnpinnedHeaderInTree()).isTrue();
+  });
+
+  testWidgets('only pinned subscriptions', (tester) async {
+    await setupStreamListPage(tester, subscriptions: [
+      eg.subscription(eg.stream(streamId: 1), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 2), pinToTop: true),
+    ]);
+    check(getItemCount()).equals(2);
+    check(isPinnedHeaderInTree()).isTrue();
+    check(isUnpinnedHeaderInTree()).isFalse();
+  });
+
+  testWidgets('only unpinned subscriptions', (tester) async {
+    await setupStreamListPage(tester, subscriptions: [
+      eg.subscription(eg.stream(streamId: 1), pinToTop: false),
+      eg.subscription(eg.stream(streamId: 2), pinToTop: false),
+    ]);
+    check(getItemCount()).equals(2);
+    check(isPinnedHeaderInTree()).isFalse();
+    check(isUnpinnedHeaderInTree()).isTrue();
+  });
+
+  testWidgets('subscription sort', (tester) async {
+    await setupStreamListPage(tester, subscriptions: [
+      eg.subscription(eg.stream(streamId: 1, name: 'd'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: false),
+      eg.subscription(eg.stream(streamId: 3, name: 'b'), pinToTop: true),
+      eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: false),
+    ]);
+    check(isPinnedHeaderInTree()).isTrue();
+    check(isUnpinnedHeaderInTree()).isTrue();
+
+    final streamListItems = tester.widgetList<SubscriptionItem>(find.byType(SubscriptionItem)).toList();
+    check(streamListItems.length).equals(4);
```

Comment:
```suggestion
    check(streamListItems).length.equals(4);
```

Using the check helpers makes the errors more informative if something breaks the test.